cloud-gateway-service/src/main/java/org/zowe/apiml/cloudgatewayservice/filters/TokenFilterFactory.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.cloudgatewayservice.filters;
+
+import lombok.EqualsAndHashCode;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.gateway.filter.GatewayFilter;
+import org.springframework.cloud.gateway.filter.GatewayFilterChain;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.server.ServerWebExchange;
+import org.zowe.apiml.cloudgatewayservice.service.InstanceInfoService;
+import org.zowe.apiml.message.core.MessageService;
+import org.zowe.apiml.zaas.ZaasTokenResponse;
+import reactor.core.publisher.Mono;
+
+import java.net.HttpCookie;
+
+
+public abstract class TokenFilterFactory extends AbstractAuthSchemeFactory<TokenFilterFactory.Config, ZaasTokenResponse, Object> {
+
+    protected TokenFilterFactory(WebClient webClient, InstanceInfoService instanceInfoService, MessageService messageService) {
+        super(Config.class, webClient, instanceInfoService, messageService);
+    }
+
+    public abstract String getEndpointUrl();
```

Comment:
```suggestion
    public abstract String getEndpointUrlPattern();
```

Reply:
Or rather consider replacing this method with a method with one argument to construct a real URL. I guess The method to return a pattern is not helpful at all (there could be different arguments required), it is much better to provide a service instance object and receive the full URL.